pkg/provider/provider.go

```diff
@@ -1,221 +1,393 @@
+				Optional:      true,
+				Sensitive:     true,
+				DefaultFunc:   schema.EnvDefaultFunc("SNOWFLAKE_PRIVATE_KEY", nil),
+				ConflictsWith: []string{"browser_auth", "password", "oauth_access_token", "private_key_path", "oauth_refresh_token"},
+			},
+			"private_key_passphrase": {
+				Type:          schema.TypeString,
+				Description:   "Supports the encryption ciphers aes-128-cbc, aes-128-gcm, aes-192-cbc, aes-192-gcm, aes-256-cbc, aes-256-gcm, and des-ede3-cbc. Can also be sourced from `SNOWFLAKE_PRIVATE_KEY_PASSPHRASE` environment variable.",
+				Optional:      true,
+				Sensitive:     true,
+				DefaultFunc:   schema.EnvDefaultFunc("SNOWFLAKE_PRIVATE_KEY_PASSPHRASE", nil),
+				ConflictsWith: []string{"browser_auth", "password", "oauth_access_token", "oauth_refresh_token"},
+			},
+			"disable_telemetry": {
+				Type:        schema.TypeBool,
+				Description: "Indicates whether to disable telemetry. Can also be sourced from the `SNOWFLAKE_DISABLE_TELEMETRY` environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_DISABLE_TELEMETRY", nil),
+			},
+			"client_request_mfa_token": {
+				Type:        schema.TypeBool,
+				Description: "When true the MFA token is cached in the credential manager. True by default in Windows/OSX. False for Linux. Can also be sourced from the `SNOWFLAKE_CLIENT_REQUEST_MFA_TOKEN` environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_CLIENT_REQUEST_MFA_TOKEN", nil),
+			},
+			"client_store_temporary_credential": {
+				Type:        schema.TypeBool,
+				Description: "When true the ID token is cached in the credential manager. True by default in Windows/OSX. False for Linux. Can also be sourced from the `SNOWFLAKE_CLIENT_STORE_TEMPORARY_CREDENTIAL` environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_CLIENT_STORE_TEMPORARY_CREDENTIAL", nil),
+			},
+			"disable_query_context_cache": {
+				Type:        schema.TypeBool,
+				Description: "Should HTAP query context cache be disabled. Can also be sourced from the `SNOWFLAKE_DISABLE_QUERY_CONTEXT_CACHE` environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_DISABLE_QUERY_CONTEXT_CACHE", nil),
+			},
+			/*
+				Feature not yet released as of latest gosnowflake release
+				https://github.com/snowflakedb/gosnowflake/blob/master/dsn.go#L103
+				"include_retry_reason": {
+					Type:        schema.TypeBool,
+					Description: "Should retried request contain retry reason",
+					Optional:    true,
+				},*/
+			"profile": {
+				Type:        schema.TypeString,
+				Description: "Sets the profile to read from ~/.snowflake/config file. Can also be sourced from the `SNOWFLAKE_PROFILE` environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_PROFILE", "default"),
+			},
+			// Deprecated attributes
```

Comment:
do we add any note somewhere else that these ones are deprecated?

Do we add it in the description of each of these parameters? ( I don't like single comment stating about deprecations, someone may add other param after this comment, and it will be misleading - that's why I would prefer having it added to every one of them)

Reply:
its enough to have the "deprecated" message, as it will show up in the documentation, and already "marks" resources as deprecated. The comment is not super important, its just for code organization I suppose.

> [tag](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/table#tag) (Block List, Deprecated) Definitions of a tag to associate with the resource. (see [below for nested schema](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/table#nestedblock--tag))